### ISSUE TITLE
Flutter Web CanvasKit Crash: `RuntimeError: table index is out of bounds`

### ISSUE THREAD

**Issue by jezell** [flutter/flutter#145134]
### Steps to reproduce

Not sure on exact repro steps besides just using our app for long enough, but it's happening almost daily for us on flutter 3.19.3 

### Expected results

Not to crash

### Actual results

CanvasKit Crashes with:

```console
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following NativeError object was thrown during a scheduler callback:
  RuntimeError: table index is out of bounds

When the exception was thrown, this was the stack:
../canvaskit/chromium/canvaskit.js 102:79                                            Cb
../canvaskit/chromium/canvaskit.js 145:437                                           delete
../lib/_engine/engine/canvaskit/native_memory.dart 88:13                             dispose
../lib/_engine/engine/canvaskit/text.dart 862:5                                      dispose
../packages/flutter/src/painting/text_painter.dart 666:29                            markNeedsLayout
../packages/flutter/src/rendering/paragraph.dart 775:5                               systemFontsDidChange
../packages/flutter/src/rendering/object.dart 4463:9                                 <fn>
../packages/flutter/src/scheduler/binding.dart 1386:7                                [_invokeFrameCallback]
../packages/flutter/src/scheduler/binding.dart 1233:11                               <fn>
../dart-sdk/lib/_internal/js_dev_runtime/private/linked_hash_map.dart 21:7           forEach
../packages/flutter/src/scheduler/binding.dart 1231:16                               handleBeginFrame
../packages/flutter/src/scheduler/binding.dart 1148:5                                [_handleBeginFrame]
../lib/_engine/engine/platform_dispatcher.dart 1361:5                                invoke1
../lib/_engine/engine/platform_dispatcher.dart 237:5                                 invokeOnBeginFrame
../lib/_engine/engine/initialization.dart 176:36                                     <fn>
../dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 550:37  _checkAndCall
../dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 555:39  dcall
2

This exception was thrown in the context of a scheduler callback. When the scheduler callback was _registered_ (as opposed to when the exception was thrown), this was the stack:
../dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 945:28  get current
../packages/flutter/src/scheduler/binding.dart 121:33                  <fn>
../packages/flutter/src/scheduler/binding.dart 123:14                  new
../packages/flutter/src/scheduler/binding.dart 605:49                  scheduleFrameCallback
../packages/flutter/src/rendering/object.dart 4455:22                  [_scheduleSystemFontsUpdate]
../packages/flutter/src/painting/binding.dart 187:7                    notifyListeners
../packages/flutter/src/painting/binding.dart 176:9                    handleSystemMessage$
../dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50  <fn>
../dart-sdk/lib/async/zone.dart 1661:54                                runUnary
../dart-sdk/lib/async/future_impl.dart 162:18                          handleValue
../dart-sdk/lib/async/future_impl.dart 838:44                          handleValueCallback
../dart-sdk/lib/async/future_impl.dart 867:13                          _propagateToListeners
../dart-sdk/lib/async/future_impl.dart 643:5                           [_completeWithValue]
../dart-sdk/lib/async/future_impl.dart 713:7                           callback
../dart-sdk/lib/async/schedule_microtask.dart 40:11                    _microtaskLoop
../dart-sdk/lib/async/schedule_microtask.dart 49:5                     _startMicrotaskLoop
../dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:7  <fn>
════════════════════════════════════════════════════════════════════════════════════════════════════
```
### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by jezell** [comment]
Still working on trying to figure out a way to repro in a small code sample, but a little challenging since it seems to be maybe some kind of GC issue inside canvaskit or the engine itself. I can manually induce a similar error on the same line by double deleting the nativeObject.

<img width="900" alt="Screenshot 2024-03-14 at 12 39 33 PM" src="https://github.com/flutter/flutter/assets/22264/0471bb40-ed20-45e5-8bcc-33bf1ad05336">

**Comment by yjbanov** [comment]
This was probably fixed by https://github.com/flutter/flutter/pull/145124. Can you please try again on the master channel?

**Comment by jezell** [comment]
@yjbanov I think it's unlikely because the emscripten objects are protected by a delete handler that itself only deletes if the object isn't deleted. I tried to cause the error to happen with a double dispose and even a double delete against the SK object, but couldn't repro it that way. Currently, I've only been able to consistently repro in OOM cases, which are surprisingly easy to hit due to the CanvasKit limiting the flutter memory to 2GB at the moment (in theory it could go up to 4GB, but by understanding is that causes issues in WebGL because WebGL doesn't use unsigned ints internally).

I'm working on a pull request that cleans up a lot of places where things aren't properly disposed in non happy path cases and provides more predictable behavior under temporary OOM situations, currently blocked on finding some time to update the build for #145380 to wrap it up. Pretty easy to add the flag to the main .gn file with the other settings, but looks like it needs to be added to all the dependencies as well to avoid linker errors.
